### PR TITLE
Add new Introduction to 2D page

### PR DESCRIPTION
**Pull request by vgezer**
This one adds a "introduction to 2D" page to the documentation which guides the user through the various elements of the UI.

Similar to my other PR: https://github.com/godotengine/godot-docs/pull/9481, but this time completely from scratch since I feel that it is missing.

### CODE REVIEW

File: tutorials/2d/introduction_to_2d.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  to `Project Settings > Display > Window > Size` and setting `Viewport Width` and `Viewport Height`.
+- **Gizmos**: Toggles the visibility of indicators of `Position` (shown with cross icon), `Lock` 
+  (shown with padlock), `Groups` (shown with two squares), and `Transformation` (shown with 
+  green and red lines).
+- **Center Selection**: The same as **Center View** button inside the viewport. Centers the selected 
+  node(s) in the view. :kbd:`F` is the default shortcut.
+- **Frame to Selection**: Similar to `Center Selection`, but also changes the zoom factor to fit the 
+  contents in the screen. :kbd:`Shift + F` is the default shortcut.
+- **Delete Guide Lines**: Deletes all guide lines from the screen. You will need to recreate them if 
+  you plan to use them later.
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:297]:
```suggestion
- **Delete Guides**: Deletes all guides from the screen. You will need to recreate them if 
  you plan to use them later. 
```
The headings here should match the actual menu options

File: tutorials/2d/introduction_to_2d.rst [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+- **Gizmos**: Toggles the visibility of indicators of `Position` (shown with cross icon), `Lock` 
+  (shown with padlock), `Groups` (shown with two squares), and `Transformation` (shown with 
+  green and red lines).
+- **Center Selection**: The same as **Center View** button inside the viewport. Centers the selected 
+  node(s) in the view. :kbd:`F` is the default shortcut.
+- **Frame to Selection**: Similar to `Center Selection`, but also changes the zoom factor to fit the 
+  contents in the screen. :kbd:`Shift + F` is the default shortcut.
+- **Delete Guide Lines**: Deletes all guide lines from the screen. You will need to recreate them if 
+  you plan to use them later. 
+- **Preview Canvas Scaling**: Toggles the preview for scaling of canvas in the editor when the zoom
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:298]:
```suggestion
- **Preview Canvas Scale**: Toggles the preview for scaling of canvas in the editor when the zoom 
```

File: tutorials/2d/introduction_to_2d.rst [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+- **Center Selection**: The same as **Center View** button inside the viewport. Centers the selected 
+  node(s) in the view. :kbd:`F` is the default shortcut.
+- **Frame to Selection**: Similar to `Center Selection`, but also changes the zoom factor to fit the 
+  contents in the screen. :kbd:`Shift + F` is the default shortcut.
+- **Delete Guide Lines**: Deletes all guide lines from the screen. You will need to recreate them if 
+  you plan to use them later. 
+- **Preview Canvas Scaling**: Toggles the preview for scaling of canvas in the editor when the zoom 
+  factor or view of the viewport changes. Useful to see how the controls will look like after scaling 
+  and moving, without running the game.
+- **Theme Preview**: Allows to choose from the available themes to change the look of control items
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:301]:
```suggestion
- **Preview Theme**: Allows to choose from the available themes to change the look of control items 
```

File: tutorials/2d/introduction_to_2d.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+Right clicking on the viewport provides two options to create a node or instantiate scene 
+in the chosen position.
+If at least one node is selected, right clicking also provides an option to move the selected 
+node(s) to this position.
+
+
+Viewport has a **View** menu which provides several options to change the look of the viewport:
+
+- **Grids**: Allows you to show grids all the time, only when using snapping, or not at all. You
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:276]:
```suggestion
- **Grid**: Allows you to show grids all the time, only when using snapping, or not at all. You 
```

File: tutorials/2d/introduction_to_2d.rst [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+
+- **Lock** (:kbd:`Ctrl + L`) the selected nodes, preventing selection and movement in the 
+  viewport. Clicking the button again (or using :kbd:`Ctrl + Shift + L`) unlocks the selected 
+  nodes. Locked nodes can only be selected in the scene tree.
+  They can easily be identified with a padlock next to their node names in the scene tree. 
+  Clicking on this padlock also unlocks the nodes.
+- **Group selected nodes** (:kbd:`Ctrl + G`). This allows selection of the root node if any 
+  of the children are selected. Using :kbd:`Ctrl + G` ungroups them. Additionally, clicking 
+  the ungroup button in the scene tree performs the same action.
+- **Skeleton Settings**: Provides options to work with Skeleton2D and Bone2D.
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:164]:
```suggestion
- **Skeleton Options**: Provides options to work with Skeleton2D and Bone2D.
```
Is the actual tooltip of this

**Review by AThousandShips — CHANGES_REQUESTED** [review]
Some immediate grammar and factual issues, will go over this again after these have been fixed to work on the general wording

**Comment by vgezer** [comment]
> Some immediate grammar and factual issues, will go over this again after these have been fixed to work on the general wording

Thanks for the effort!

**Comment by AThousandShips** [comment]
Will do a final pass over this today or this weekend

File: tutorials/2d/introduction_to_2d.rst [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+  and moving, without running the game.
+- **Preview Theme**: Allows to choose from the available themes to change the look of control items 
+  in the editor, without requiring to run the game.
+
+
+Node2D and Control node
+~~~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Node2D <class_Node2D>` is the base node for 2D. :ref:`Control <class_Control>` is the base node 
+for everything GUI. For the 3D, the Godot uses the :ref:`Node3D <class_Node3D>` node.
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:309]:
```suggestion
for everything GUI. For 3D, Godot uses the :ref:`Node3D <class_Node3D>` node.
```

File: tutorials/2d/introduction_to_2d.rst [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+node(s) to this position.
+
+
+Viewport has a **View** menu which provides several options to change the look of the viewport:
+
+- **Grid**: Allows you to show grids all the time, only when using snapping, or not at all. You 
+  can also toggle them with the provided option.
+- **Show Helpers**: Toggles the temporary display of an outline of the node, with the previous 
+  transform properties (position, scaling, or rotation) if a transform operation has been 
+  initiated. For the `Control` nodes, it also shows the sizing parameters. Useful to see the deltas.
```
Comment (AThousandShips) [inline tutorials/2d/introduction_to_2d.rst:280]:
```suggestion
  initiated. For `Control` nodes, it also shows the sizing parameters. Useful to see the deltas.
```

**Review by AThousandShips — APPROVED** [review]
Might benefit from some tweaks and polish down the line but with these fixes it should be good to go for my part

File: tutorials/2d/introduction_to_2d.rst [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+- **Select Mode** (:kbd:`Q`): Allows selection of nodes in the viewport. Left clicking on a node 
+  in the viewport selects one.
+  Left clicking and dragging a rectangle selects all nodes within the rectangle's boundaries,
+  once released.
+  Holding :kbd:`Shift` while selecting adds more nodes to the selection.
+  Clicking on a selected node while holding :kbd:`Shift` deselects the node.
+  In this mode, you can drag the selected node(s) to move, press :kbd:`Ctrl` to switch to the 
+  rotation mode temporarily, or use the red circles to scale it. If multiple nodes are 
+  selected, only movement and rotation are possible. In this mode, rotation and scaling 
+  will not use the snapping options if snapping enabled.
```
[the remark this answers is not in the document]
Comment (AThousandShips) [reply tutorials/2d/introduction_to_2d.rst:71]:
```suggestion
  will not use the snapping options if snapping is enabled.
```

File: tutorials/2d/introduction_to_2d.rst [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+**Center View** centers the selected node(s) in the screen. Useful if you have a large scene 
+with many nodes, and want to see the node selected in the scene tree.
+Next to it, are the zoom controls. **-** zooms out, **+** zooms in, and clicking on the number 
+with percentage defaults to 100%.
+Alternatively, you can use middle-mouse scrolling to zoom in (scroll up) and out (scroll down).
+
+The black bars at the viewport's left and top are the **rulers**. You can use them to 
+orient yourself in the viewport.
+By default, the rulers will display the pixel coordinates of the viewport, numbered at 
+each 100 pixels. Changing zoom factor will also increase or decrease the scaling of these
```
[the remark this answers is not in the document]
Comment (vgezer) [reply tutorials/2d/introduction_to_2d.rst:234]: Proposal: By default, the rulers will display the pixel coordinates of the viewport, numbered at 
100 pixel steps. Changing the zoom factor will change the shown values.

File: tutorials/2d/introduction_to_2d.rst [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+
+If you have at least one node in the scene, clicking on the black area of the horizontal or 
+vertical ruler and dragging the mouse toward the viewport will create a purple guide, also showing 
+its global position in the other axis. Releasing the mouse will leave it in that position. If you use 
+the gray square located at the intersection of both the horizontal and vertical rulers, you will 
+create a horizontal and a vertical line at the same time. Horizontal guides can be repositioned 
+by dragging them inside the vertical ruler. Once you hover your mouse over a line inside the 
+vertical ruler, your cursor will change. Similarly, vertical guides can be repositioned by 
+dragging inside the horizontal ruler. To remove a guide, move it all the way back to the 
+respective ruler.
```
[the remark this answers is not in the document]
Comment (vgezer) [reply tutorials/2d/introduction_to_2d.rst]: Proposal: If you have at least one node in the scene, you can create guides by dragging from the horizontal 
or vertical ruler towards the viewport. A purple guide will appear, showing its position, and will 
remain there when you release the mouse. You can create both horizontal and vertical guides 
simultaneously by dragging from the gray square at the rulers' intersection. Guides can be 
repositioned by dragging them back to their respective rulers, and they can be removed by 
dragging them all the way back to the ruler.

File: tutorials/2d/introduction_to_2d.rst [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+
+.. figure:: img/2d_platformer_demo.webp
+   
+   2D Platformer Demo available on the Asset Library.
+
+This page will show you the 2D workspace and how you can get to know it.
+
+.. tip:: If you would like to get an introduction to 3D, see :ref:`doc_introduction_to_3d`.
+
+2D workspace
```
Comment (tetrapod00) [inline tutorials/2d/introduction_to_2d.rst:24]: Note that it's [not completely clear](https://github.com/godotengine/godot-docs/issues/5731) whether the preferred term is:
- "2D Workspace", as used here and in some other places in the docs
- "2D main screen", or "2D Editor", as used in the editor itself.

Godot_v4 4-dev4_win64_rezKl8J8lA

I'd be very happy using the terms "Workspaces" and "2D Workspace", but be aware we are setting some precedent here.

File: tutorials/2d/introduction_to_2d.rst [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+~~~~~~~~~~~~
+
+You will use the 2D workspace to work with 2D scenes, design levels, or create user 
+interfaces.
+You can use the workspace selector located in Godot's top edge at the middle, or choose a 2D 
+related node from the scene tree to switch to the 2D editor:
```
Comment (tetrapod00) [inline tutorials/2d/introduction_to_2d.rst:30]:
```suggestion
To switch to the 2D workspace, you can either select a 2D node from the scene tree, or use the workspace selector located at the top edge of the editor:
```
I think if we pick "2D workspace" as the term, it should be consistently used here? Also mildy reworded for clarity

File: tutorials/2d/introduction_to_2d.rst [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+  and moving, without running the game.
+- **Preview Theme**: Allows to choose from the available themes to change the look of control items 
+  in the editor, without requiring to run the game.
+
+
+Node2D and Control node
+~~~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Node2D <class_Node2D>` is the base node for 2D. :ref:`Control <class_Control>` is the base node 
+for everything GUI. For 3D, Godot uses the :ref:`Node3D <class_Node3D>` node.
```
Comment (tetrapod00) [inline tutorials/2d/introduction_to_2d.rst:305]:
```suggestion
:ref:`CanvasItem` is the base node for 2D. :ref:`Node2D <class_Node2D>` is the base node
for 2D game objects, and :ref:`Control <class_Control>` is the base node 
for everything GUI. For 3D, Godot uses the :ref:`Node3D <class_Node3D>` node.
```
As-is, this seems misleading. `CanvasItem` is the base 2D node, from the [CanvasItem docs:](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#description)
> Abstract base class for everything in 2D space. Canvas items are laid out in a tree; children inherit and extend their parent's transform. CanvasItem is extended by [Control](https://docs.godotengine.org/en/stable/classes/class_control.html#class-control) for GUI-related nodes, and by [Node2D](https://docs.godotengine.org/en/stable/classes/class_node2d.html#class-node2d) for 2D game objects.

But I'm not sure if my suggestion is the best way to clarify, I'm not a 2D expert.

**Review by tetrapod00 — COMMENTED** [review]
This looks really good. I think a couple things need changes but I could also see merging this as-is, honestly.

File: tutorials/2d/introduction_to_2d.rst [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+
+Welcome to the exciting world of 2D game development with Godot! Whether you're a 
+seasoned developer or just starting out, Godot offers a friendly and intuitive environment 
+to bring your game ideas to life. With its powerful yet accessible features, you'll have 
+the freedom to unleash your creativity and create captivating gaming experiences. From 
+designing unique characters and crafting immersive worlds to implementing engaging 
+gameplay mechanics, Godot provides a comprehensive set of tools and resources to support 
+your game development journey. So, get ready to embark on an adventure filled with endless 
+possibilities and let your imagination soar as you dive into the wonderful realm of 2D game 
+development with Godot!
```
Comment (tetrapod00) [inline tutorials/2d/introduction_to_2d.rst:14]:
```suggestion
Welcome to 2D game development with Godot! Whether you're a seasoned developer or
just starting out, Godot offers a friendly and intuitive environment to bring your game
ideas to life. Godot has dedicated 2D rendering and physics engines, along with many
features designed just for 2D. You can craft unique worlds with the TileMap system,
animate your characters with 2D sprite animation or Cutout animation, and use 2D
lighting to dynamically light your scenes. There is also a robust 2D particle system to
create visual effects, with custom shader support. So, get ready to dive into the wonderful
realm of 2D game development with Godot!
```
Applied most of the feedback here, while trying to keep a little bit of the original tone.

Optionally each of these features could be a link to the appropriate page, but I don't think that's necessary in an intro paragraph.

File: tutorials/2d/introduction_to_2d.rst [hunk 15]
```diff
@@ -0,0 +1,316 @@
...
+  An added node has its rotation pivot at ``x: 0``, ``y: 0``, by default, with
+  exceptions. For example, the default pivot for a :ref:`Sprite2D <class_Sprite2D>` is its 
+  center if the ``centered`` property is set to ``true``. If you would like to change the 
+  rotation pivot of a node, click this button and choose a new location by left clicking. 
+  The node rotates considering this point. If you have multiple nodes selected, this icon 
+  will add a temporary pivot to be used commonly by all selected nodes. Pressing :kbd:`Shift` 
+  and clicking this button will create the pivot at the center of selected nodes. If any of 
+  the snap options are enabled, the pivot will also snap to them it when dragged.
+- **Pan Mode**: Allows you to navigate in the viewport without accidentally selecting any nodes.
+  You can also use :kbd:`G` (or :kbd:`Space`) and drag with the left mouse button to do the same.
```
Comment (tetrapod00) [inline tutorials/2d/introduction_to_2d.rst:91]:
```suggestion
- **Pan Mode** (:kbd:`G`): Allows you to navigate in the viewport without accidentally selecting any nodes.
  In other modes, you can also hold :kbd:`Space` and drag with the left mouse button to do the same.
```

**Review by tetrapod00 — COMMENTED** [review]
If the suggestions are applied this looks great to me!

I read through the whole page and checked that the tools and hotkeys are mostly accurate, though it's possible I missed something.

File: tutorials/2d/introduction_to_2d.rst [hunk 16]
```diff
@@ -0,0 +1,315 @@
...
+  factor or view of the viewport changes. Useful to see how the controls will look like after scaling 
+  and moving, without running the game.
+- **Preview Theme**: Allows to choose from the available themes to change the look of control items 
+  in the editor, without requiring to run the game.
+
+
+Node2D and Control node
+~~~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`CanvasItem` is the base node for 2D. :ref:`Node2D <class_Node2D>` is the base node
```
Comment (tetrapod00) [inline tutorials/2d/introduction_to_2d.rst:302]:
```suggestion
:ref:`CanvasItem <class_CanvasItem>` is the base node for 2D. :ref:`Node2D <class_Node2D>` is the base node
```
Sorry, I gave you bad syntax in the other suggestion. This should fix the CI.

Comment (vgezer) [reply tutorials/2d/introduction_to_2d.rst:302]: I will merge all and make the changes you suggested above as well. Thanks!

**Review by tetrapod00 — APPROVED** [review]
LGTM. Squashing the commits would be nice, if you can.

File: tutorials/2d/introduction_to_2d.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (vgezer) [reply tutorials/2d/introduction_to_2d.rst]: Another shot.

File: tutorials/2d/introduction_to_2d.rst (hunk 11 above)
Comment (vgezer) [reply tutorials/2d/introduction_to_2d.rst:22]: Is this then fine? 2D Workspace?

Comment (tetrapod00) [reply tutorials/2d/introduction_to_2d.rst:22]: Yeah, this is fine as-is, you don't need to make changes.

I'm in favor of calling these workspaces. I just wanted to point out that it's a bit of an open discussion, and let anyone else comment if they cared.

**Comment by tetrapod00** [comment]
Thank you, great work!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.3 in https://github.com/godotengine/godot-docs/pull/10363.

**Pull request merged**